### PR TITLE
Sf 1088 design system upgrade tabs

### PR DESCRIPTION
**Pull request by leeshuzheng**

### CODE REVIEW

File: src/components/atoms/SubTab/SubTab.tsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import clsx from 'clsx';
+import { SubTabProps } from './types';
+
+export const SubTab = ({ text, active }: SubTabProps) => {
+    return (
+        <span
+            className={clsx(
+                'group flex h-full w-full items-center justify-center rounded-md px-[5.88px] py-1.5 text-xs font-semibold leading-4 duration-300 hover:opacity-100 hover:ease-in-out laptop:px-3 laptop:leading-5',
```
Comment (as472) [inline src/components/atoms/SubTab/SubTab.tsx:8]: please use the respective typography

Comment (as472) [inline src/components/atoms/SubTab/SubTab.tsx:8]: hover classes don't do anything here anymore

File: src/components/atoms/SubTab/SubTab.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+} as Meta<typeof SubTab>;
+
+const Template: StoryFn<typeof SubTab> = args => (
+    <div className='max-w-[175px]'>
+        <SubTab {...args} />
+    </div>
+);
+
+export const Default = Template.bind({});
+Default.args = {};
```
Comment (as472) [inline src/components/atoms/SubTab/SubTab.stories.tsx:25]: we do not need line 25

File: src/components/atoms/index.ts [hunk 3]
```diff
@@ -1,3 +1,4 @@
+export * from '../molecules/TabGroup';
```
Comment (as472) [inline src/components/atoms/index.ts:1]: we should not be exporting a molecule from atoms

File: src/components/atoms/Tab/Tab.tsx [hunk 4]
```diff
@@ -0,0 +1,83 @@
+import clsx from 'clsx';
+import { HighlightChip } from '../HighlightChip';
```
Comment (as472) [inline src/components/atoms/Tab/Tab.tsx:2]: please import from src/components/atoms

File: src/components/molecules/HistoricalChart/HistoricalChart.tsx [hunk 5]
```diff
@@ -118,14 +117,6 @@ export function HistoricalChart({
                 ...volumeData,
             } as Omit<ITradingData, 'vol'> & { value: number };
 
-            const date = new Date(Number(candleData?.time) * 1000);
```
Comment (as472) [inline src/components/molecules/HistoricalChart/HistoricalChart.tsx:121]: looks like unrelated to tabs/sub tabs. Could you give some background for this change?

**Review by as472 — COMMENTED** [review]
@leeshuzheng you will need to take a pull from develop as OrderBookWidget is now deleted.

File: src/components/organisms/LendingCard/LendingCard.tsx [hunk 6]
```diff
@@ -154,10 +155,15 @@ export const LendingCard = ({
 
     const { data: availableToBorrow } = useBorrowableAmount(address, currency);
 
+    const orderSideOptions = Object.values(OrderSideMap).map(option => ({
```
Comment (as472) [inline src/components/organisms/LendingCard/LendingCard.tsx:158]: this is a fixed object (independent of the component). Best to take it out to not create it on every rerender.

File: src/components/molecules/SubtabGroup/SubtabGroup.tsx [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+    options: string[];
+    selectedOption: string;
+    handleClick: (option: string) => void;
+}) => {
+    return (
+        <RadioGroup
+            value={selectedOption}
+            onChange={handleClick}
+            as='div'
+            className='flex flex-row items-center gap-1 rounded-lg bg-tabBg px-1.5 py-[5px] laptop:gap-2 laptop:p-2'
```
Comment (as472) [inline src/components/molecules/SubtabGroup/SubtabGroup.tsx:18]: the background might be neutral-800

File: src/components/molecules/SubtabGroup/SubtabGroup.stories.tsx [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+            />
+        </div>
+    );
+};
+
+export const Default = Template.bind({});
+
+export const GroupOfThree = Template.bind({});
+GroupOfThree.args = {
+    options: ['Tab label', 'Label', 'Label 2'],
```
Comment (as472) [inline src/components/molecules/SubtabGroup/SubtabGroup.stories.tsx:45]: might need to pass the selectedOption as it seems it is still referring to the selectedOption of the default story

File: src/components/molecules/HistoricalChart/HistoricalChart.tsx (hunk 5 above)
Comment (leeshuzheng) [reply src/components/molecules/HistoricalChart/HistoricalChart.tsx:121]: @as472 Yeah these are unrelated to tabs/subtabs. I see that the width for Historical Chart is smaller on small desktop screens, and that @benk-sf removed the hover time display on the chart: https://www.figma.com/design/o2wjskNCdkt65haAgWPxo6/Product-UI-(Main)?node-id=15841-65963&m=dev

On our current implementation, the hover time/date is being cut off because of insufficient space, so I thought this would be a good time to remove that

File: src/components/molecules/SubtabGroup/SubtabGroup.tsx (hunk 7 above)
Comment (leeshuzheng) [reply src/components/molecules/SubtabGroup/SubtabGroup.tsx:18]: @as472 Yeah that is the case in the design system file, but in the hyperliquid ticket design it's updated: https://www.figma.com/design/o2wjskNCdkt65haAgWPxo6/Product-UI-(Main)?node-id=15836-33212&m=dev

Will inform @benk-sf to update it in design system

File: src/components/molecules/SubtabGroup/SubtabGroup.stories.tsx (hunk 8 above)
Comment (leeshuzheng) [reply src/components/molecules/SubtabGroup/SubtabGroup.stories.tsx:45]: Got it. Will add that in

File: src/components/organisms/LendingCard/LendingCard.tsx (hunk 6 above)
Comment (leeshuzheng) [reply src/components/organisms/LendingCard/LendingCard.tsx:158]: Ok I will place it before the component is defined

File: src/components/atoms/SubTab/SubTab.tsx (hunk 1 above)
Comment (leeshuzheng) [reply src/components/atoms/SubTab/SubTab.tsx:8]: @as472 we don't have typography for mobile in figma: https://www.figma.com/design/YIBsk1ihFbPlDb8XldGj1T/SF-Design-System?node-id=614-6853&m=dev

so I'll use the tailwind classes for mobile, then do a laptop:typography-caption-2 with it (for desktop)

Comment (as472) [reply src/components/atoms/SubTab/SubTab.tsx]: use the new classes to define the typography.
typography-desktop-body-5

File: src/components/molecules/SubtabGroup/SubtabGroup.tsx (hunk 7 above)
Comment (as472) [reply src/components/molecules/SubtabGroup/SubtabGroup.tsx:18]: okay. sure

File: src/components/organisms/LendingCard/constants.ts [hunk 9]
```diff
@@ -0,0 +1,7 @@
+import { TabVariant } from 'src/components/atoms';
+import { OrderSideMap } from 'src/types';
+
+export const orderSideOptions = Object.values(OrderSideMap).map(option => ({
+    text: option,
+    variant: 'Blue' as TabVariant,
```
Comment (as472) [inline src/components/organisms/LendingCard/constants.ts:6]: is there an issue if we write as TabVariant.Blue

File: src/components/organisms/AdvancedLendingOrderCard/AdvancedLendingOrderCard.tsx [hunk 10]
```diff
@@ -312,77 +321,61 @@ export function AdvancedLendingOrderCard({
         [setIsShowingAll]
     );
 
+    const orderSideOptions = Object.values(OrderSideMap).map(option => ({
```
Comment (as472) [inline src/components/organisms/AdvancedLendingOrderCard/AdvancedLendingOrderCard.tsx:324]: this can move outside the component as well
